controllers/operatorpolicy_status.go

```diff
@@ -1059,10 +1104,20 @@ func existingInstallPlanObj(ip client.Object, phase string) policyv1.RelatedObje
 	}
 
 	switch phase {
+	case string(operatorv1alpha1.InstallPlanPhaseInstalling):
+		// Check policy.spec.statusConfig.upgradesAvailable to determine `compliant`.
+		if complianceConfig != "Compliant" {
+			relObj.Compliant = string(policyv1.NonCompliant)
+		} else {
+			relObj.Compliant = string(policyv1.Compliant)
+		}
 	case string(operatorv1alpha1.InstallPlanPhaseRequiresApproval):
-		// FUTURE: check policy.spec.statusConfig.upgradesAvailable to determine `compliant`.
-		// For now, assume it is set to 'NonCompliant'
-		relObj.Compliant = string(policyv1.NonCompliant)
+		// Check policy.spec.statusConfig.upgradesAvailable to determine `compliant`.
+		if complianceConfig != "Compliant" {
+			relObj.Compliant = string(policyv1.NonCompliant)
+		} else {
+			relObj.Compliant = string(policyv1.Compliant)
+		}
```

Comment:
Should these also append to the reason, like we do for the condition messages?

Reply:
I think this makes sense, I'll add it to the other relatedObj `reason` messages as well for consistency.